File: RestfulObjects Server/RestfulObjects.Test.Data/RestDataRepository.cs
// Copyright Naked Objects Group Ltd, 45 Station Road, Henley on Thames, UK, RG9 1AT
// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. 
// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and limitations under the License.

using System.Linq;
using NakedObjects.Services;

namespace RestfulObjects.Test.Data {
    //[DisplayName("")]
    public class RestDataRepository : AbstractFactoryAndRepository {
        #region Injected Services

        // This region should contain properties to hold references to any services required by the
        // object.  Use the 'injs' shortcut to add a new service.

        #endregion

        // 'fact' shortcut to add a factory method, 
        // 'alli' for an all-instances method
        // 'find' for a method to find a single object by query
        // 'list' for a method to return a list of objects matching a query

        public MostSimple AzContributedAction(WithActionObject withAction) {
            return Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
        }

        public MostSimple AzContributedActionOnBaseClass(WithAction withAction) {
            return Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
        }

        public MostSimple AzContributedActionWithRefParm(WithActionObject withAction, WithActionObject withOtherAction) {
            return Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
        }

        public MostSimple AzContributedActionWithValueParm(WithActionObject withAction, string parm) {
            return Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
        }

        public MostSimple CreateTransientMostSimple() {
            return Container.NewTransientInstance<MostSimple>();
        }

        public WithValue CreateTransientWithValue() {
            var withValue = Container.NewTransientInstance<WithValue>();

            withValue.AValue = 102;
            withValue.AChoicesValue = 3;
            withValue.AConditionalChoicesValue = 3;
            withValue.ADisabledValue = 103;
            withValue.AStringValue = "one hundred four";
            withValue.AHiddenValue = 105;

            return withValue;
        }

        public WithReference CreateTransientWithReference() {
            var wr = Container.NewTransientInstance<WithReference>();

            wr.AReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            //wr.AHiddenReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            wr.AChoicesReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            wr.AConditionalChoicesReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            wr.ADisabledReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            wr.AnEagerReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            wr.AnAutoCompleteReference = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);

            return wr;
        }

        public WithCollection CreateTransientWithCollection() {
            var twc = Container.NewTransientInstance<WithCollection>();
            var ms1 = Container.Instances<MostSimple>().Single(ms => ms.Id == 1);
            var ms2 = Container.Instances<MostSimple>().Single(ms => ms.Id == 2);

            twc.ACollection.Add(ms1);
            twc.ACollection.Add(ms2);

            twc.ASet.Add(ms1);
            twc.ASet.Add(ms2);

            twc.ACollectionViewModels.Add(new MostSimpleViewModel() {Id = 1});
            twc.ACollectionViewModels.Add(new MostSimpleViewModel() {Id = 2});

            twc.ADisabledCollection.Add(ms1);
            twc.ADisabledCollection.Add(ms2);

            twc.AHiddenCollection.Add(ms1);
            twc.AHiddenCollection.Add(ms2);

            twc.AnEagerCollection.Add(ms1);
            twc.AnEagerCollection.Add(ms2);

            return twc;
        }
    }
}

File: RestfulObjects Server/RestfulObjects.Test.Data.Sdm2/Services/RestDataRepository.vb
Imports System.Linq
Imports sdm.systems.application.container


Namespace RestfulObjects.Test.Data
    Public Class RestDataRepository
        Inherits SdmObject
        Implements IAccessibleThroughAware
        Implements IRestDataRepository

        Public Overrides Function title() As Title
            Dim t As New Title
            t.append("Rest Data Repository")
            Return t
        End Function

        Public Function actionAzContributedAction(withAction As WithActionObject) As MostSimple _
            Implements IRestDataRepository.AzContributedAction
            Return Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().First()
        End Function

        Public Function actionAzContributedActionOnBaseClass(withAction As WithAction) As MostSimple
            Return Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().First()
        End Function

        Public Function actionAzContributedActionWithRefParm(withAction As WithActionObject) As MostSimple _
            Implements IRestDataRepository.AzContributedActionWithRefParm
            Return Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().First()
        End Function

        Public Function actionAzContributedActionWithValueParm(withAction As WithActionObject) As MostSimple _
            Implements IRestDataRepository.AzContributedActionWithValueParm
            Return Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().First()
        End Function


        Public Function actionCreateTransientMostSimple() As MostSimple
            Return Container.createTransientInstance(GetType(MostSimple))
        End Function

        Public Function actionCreateTransientWithValue() As WithValue
            'Return Container.createTransientInstance(GetType(WithValue))

            Dim withValue As WithValue
            withValue = Container.createTransientInstance(GetType(WithValue))

            withValue.AValue.setValue(102)
            withValue.AChoicesValue.setValue(3)
            withValue.ADisabledValue.setValue(103)
            withValue.AStringValue.setValue("one hundred four")
            withValue.AHiddenValue.setValue(105)

            Return withValue
        End Function

        Public Function actionCreateTransientWithReference() As WithReference
            Dim wr As WithReference
            wr = Container.createTransientInstance(GetType(WithReference))

            wr.AReference =
                Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().Single(
                    Function(ms) ms.Id.longValue() = 1)
            wr.AChoicesReference =
                Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().Single(
                    Function(ms) ms.Id.longValue() = 1)
            wr.ADisabledReference =
                Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().Single(
                    Function(ms) ms.Id.longValue() = 1)
            wr.AHiddenReference =
                Container.allInstances(GetType(MostSimple)).Cast (Of MostSimple)().Single(
                    Function(ms) ms.Id.longValue() = 1)

            Return wr
        End Function

        Public Function actionCreateTransientWithCollection() As WithCollection
            Return Container.createTransientInstance(GetType(WithCollection))
        End Function


        Public Function AwareType(ByVal accessibleType As Type) As Type _
            Implements IAccessibleThroughAware.AwareType
            Return GetType(IWithActionRepositoryAware)
        End Function
    End Class
End Namespace
